src/main/java/org/cytoscape/rest/internal/resource/NetworkResource.java

```diff
@@ -1538,8 +1538,11 @@ private final Map<CyNetworkView, VisualStyle> addNetwork(final CyNetwork[] netwo
 				CySubNetwork subnet = (CySubNetwork) networks[0];
 				final CyRootNetwork rootNet = subnet.getRootNetwork();
 				String rootNetName = rootNet.getRow(rootNet).get(CyNetwork.NAME, String.class);
-				rootNet.getRow(rootNet).set(CyNetwork.NAME, collectionName);
-				if (rootNetName == null || rootNetName.trim().length() == 0) {
+				if ( collectionName != null && collectionName.trim().length() > 0) {
```

Comment:
If you want you can use `!collectionName.isBlank()` instead of `collectionName.trim().length() > 0`. Its slightly shorter.